packages/replay-next/src/suspense/PointStackCache.ts

```diff
@@ -0,0 +1,37 @@
+import { ExecutionPoint, PointStackFrame } from "@replayio/protocol";
+import { IntervalCache, createIntervalCache } from "suspense";
+
+import { ReplayClientInterface } from "shared/client/types";
+
+import { updateMappedLocation } from "./PauseCache";
+import { sourcesByIdCache } from "./SourcesCache";
+
+export const pointStackCache: IntervalCache<
+  number,
+  [replayClient: ReplayClientInterface, point: ExecutionPoint],
+  PointStackFrame & { index: number }
+> = createIntervalCache({
+  debugLabel: "pointStackCache",
+  getPointForValue: pointStackFrame => pointStackFrame.index,
+  getKey: (client, point) => point,
+  load: async (start, end, client, point) => {
+    const pointStack = await client.getPointStack(point, end + 1);
+    const sources = await sourcesByIdCache.readAsync(client);
+    return pointStack.slice(start).map((frame, index) => {
+      updateMappedLocation(sources, frame.functionLocation);
+      if (frame.point.frame) {
+        updateMappedLocation(sources, frame.point.frame);
```

Comment:
I don't think I understand the relationship between these two. The Protocol docs say:

> `PointStackFrame`:
> * `functionLocation` – The location of the function containing the point
> * `point.frame` – Location of the topmost frame, omitted if there are no frames on stack